pkg/pod/status.go

```diff
@@ -175,12 +191,14 @@ func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses [
 					trs.TaskRunResults = append(trs.TaskRunResults, taskResults...)
 					trs.ResourcesResult = append(trs.ResourcesResult, pipelineResourceResults...)
 				}
-				msg, err = createMessageFromResults(filteredResults)
-				if err != nil {
-					logger.Errorf("%v", err)
-					merr = multierror.Append(merr, err)
-				} else {
-					s.State.Terminated.Message = msg
+				if len(sidecarLogResults) == 0 {
+					msg, err = createMessageFromResults(filteredResults)
+					if err != nil {
+						logger.Errorf("%v", err)
+						merr = multierror.Append(merr, err)
+					} else {
+						s.State.Terminated.Message = msg
+					}
```

Comment:
this check based on length of `sidecarLogResults` is confusing after already combining results from both extraction methods

Reply:
For context, this was done because the CI tests run the same test environment for all example tests and e2e tests. As a result, certain alpha features were failing this test because sidecar-logs was activated. The solution for now is to disable `sidecar-logs` for all alpha e2e and example tests. Only trigger it for the specific e2e tests. We will figure out how to handle examples next.